src/root/views/GoUI/components/Button/index.tsx

```diff
@@ -1,190 +1,123 @@
-    childrenClassName?: string;
-    actionsClassName?: string;
-    disabled?: boolean;
-    buttonWithDescription?: String;
-    name: N;
-    onClick?: (name: N, e: React.MouseEvent<HTMLButtonElement>) => void;
+export interface Props<N> extends RawButtonProps<N> {
+  variant?: ButtonVariant;
+  actions?: React.ReactNode;
+  actionsClassName?: string;
+  childrenClassName?: string;
+  disabled?: boolean;
+  icons?: React.ReactNode;
+  iconsClassName?: string;
+  name: N;
+  onClick?: (name: N, e: React.MouseEvent<HTMLButtonElement>) => void;
 }
 
-type ButtonFeatureKeys =
-    'variant'
-    | 'className'
-    | 'actionsClassName'
-    | 'iconsClassName'
-    | 'childrenClassName'
-    | 'children'
-    | 'icons'
-    | 'actions'
-    | 'disabled'
-    | 'buttonWithDescription'
-
-export type ButtonFeatureProps<N> = Pick<Props<N>, ButtonFeatureKeys>;
+export type ButtonFeatureProps<N> = Omit<Props<N>, 'name' | 'onClick'>;
 export function useButtonFeatures<N>(
-    props: ButtonFeatureProps<N>,
+  props: ButtonFeatureProps<N>,
 ) {
-    const {
-        variant = 'primary',
-        className,
-        actionsClassName,
-        iconsClassName,
-        childrenClassName,
-        disabled,
-        children,
-        icons,
-        actions,
-        buttonWithDescription,
-    } = props;
+  const {
+    actions,
+    actionsClassName,
+    children,
+    childrenClassName,
+    className,
+    disabled,
+    icons,
+    iconsClassName,
+    variant = 'primary',
+  } = props;
 
-    const buttonClassName = _cs(
-        styles.button,
-        variant,
-        buttonVariantToStyleMap[variant] ?? styles.primary,
-        disabled && styles.disabled,
-        className,
-    );
+  const buttonClassName = _cs(
+    styles.button,
+    buttonVariantToStyleMap[variant],
+    disabled && styles.disabled,
+    className,
+  );
 
-    const buttonChildren = (
-        <>
-            {icons && (
-                <div className={_cs(iconsClassName,
-                    (variant === 'navigateTop' ? styles.navigateTopIcon : styles.icons),
-                )}>
-                    {icons}
-                </div>
-            )}
-            {children && (
-                <div className={_cs(childrenClassName, styles.children)}>
-                    {children}
-                </div>
-            )}
-            {actions && (
-                <div className={_cs(actionsClassName, styles.actions)}>
-                    {actions}
-                </div>
-            )}
-        </>
-    );
+  const buttonChildren = (
+    <>
+      {icons && (
+        <div className={_cs(iconsClassName, styles.icon)}>
+          {icons}
+        </div>
+      )}
+      {children && (
+        <div className={_cs(childrenClassName, styles.children)}>
+          {children}
+        </div>
+      )}
+      {actions && (
+        <div className={_cs(actionsClassName, styles.actions)}>
+          {actions}
+        </div>
+      )}
+    </>
+  );
```

Comment:
can we use ElementFragment or useBasicLayout ?

Reply:
I have used `useBasicLayout` hooks for now.